collie/interactions/datasets.py

```diff
@@ -383,66 +417,76 @@ def __getitem__(self, index: Union[int, Iterable[int]]) -> (
         item_id = self.mat.col[index]
         # rating = self.mat.data[index]  # not needed for any loss currently implemented
 
-        negative_item_ids_array = self._negative_sample(user_id)
+        if self.negative_sample_type == 'item':
+            negative_ids_array = self._negative_sample(user_id)
+        elif self.negative_sample_type == 'user':
+            negative_ids_array = self._negative_sample(item_id)
 
-        return (user_id, item_id), negative_item_ids_array
+        return (user_id, item_id), negative_ids_array
+
+    def _negative_sample(self, id: Union[int, np.array]) -> np.array:
```

Comment:
I know this sucks, but we should avoid using a variable name `id` since it's a built-in Python function: https://docs.python.org/3/library/functions.html#id

Reply:
Do you hate `_id`?

Reply:
I am sorry to say it but I do hate `_id`. Unfortunately, I cannot make a suggestion of another name.

Reply:
😠